src/mdx/expandVersionedUrl.js

```diff
@@ -1,27 +1,71 @@
-const tokens = [
-  {
-    token: "$docs$",
-    rules: [
-      { match: "/optimize/", expandTo: "/docs/next" },
-      // these other mappings are currently wrong because there are no optimize versions yet...but eventually, this is what we'll want.
-      { match: "/optimize_versioned_docs/version-3.8.0/", expandTo: "/docs" },
-      {
-        match: "/optimize_versioned_docs/version-3.7.0/",
-        expandTo: "/docs/1.3",
-      },
-    ],
-  },
+// @ts-check
+
+/**
+ * @typedef {object} VersionMapping
+ * @property {string} docsVersion
+ * @property {string} optimizeVersion
+ */
+
+/** @type {Array<VersionMapping>} */
```

Comment:
These type definitions were useful for me when writing the code. They aren't actually checked in the build process because we don't have docusaurus configured for typescript (yet!), but in VS Code at least, they spit out errors when things aren't passing. 

I can remove them if we think they're noisy.

Reply:
I don't have strong feelings about this, but please leave a brief comment in the code.

Reply:
Added in most recent commit!